web/html/src/manager/minion/ansible/accordion-path-content.tsx

```diff
@@ -0,0 +1,182 @@
+import * as React from "react";
+import { AnsiblePath } from "./ansible-path-type";
+import Network from "utils/network";
+import { Messages, Utils } from "components/messages";
+import { Loading } from "components/utils/Loading";
+import { AceEditor } from "components/ace-editor";
+
+type PropsType = {
+  path: AnsiblePath;
+};
+
+type StateType = {
+  open: boolean; 
+  content: any;
+  errors: string[];
+  loading: boolean;
+};
+
+function getURL(path: AnsiblePath) {
+  let baseUrl: string;
+  if (path.type === "playbook") {
+    baseUrl = "/rhn/manager/api/systems/details/ansible/discover-playbooks/";
+  }
+  else {
+    baseUrl = "/rhn/manager/api/systems/details/ansible/introspect-inventory/";
+  }
+  return baseUrl + path.id;
+}
+
+interface PlaybookDetails {
+  path: AnsiblePath,
+  fullPath: string,
+  customInventory?: string
+  name: string,
+}
+
+class AccordionPathContent extends React.Component<PropsType, StateType> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      open: false,
+      content: null,
+      errors: [],
+      loading: false,
+    };
+  }
+
+  onToggle() {
+    const path: AnsiblePath = this.props.path;
+    if (!this.state.open) {
+      if (this.state.content === null) {
+        this.setState({ loading: true });
+        Network.get(getURL(path))
```

Comment:
FYI this will lead to issues if the user a-la double-clicks on the header and then the response will arrive after another change state has already triggered. We have issues like this in a lot of places though so I don't think it's worth solving here right now, we can tackle that somewhere in the undefined future.